common/src/main/java/com/microsoft/identity/common/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -334,6 +350,165 @@ class NativeAuthMsalController : BaseNativeAuthController() {
+                performSignUpStartUsingPasswordRequest(
+                    oAuth2Strategy = oAuth2Strategy,
+                    parameters = (parameters as SignUpStartUsingPasswordCommandParameters)
+                )
+            } else {
+                performSignUpStartRequest(
+                    oAuth2Strategy = oAuth2Strategy,
+                    parameters = (parameters as SignUpStartCommandParameters)
+                )
+            }
+            return when (signUpStartApiResult) {
+                is SignUpStartApiResult.VerificationRequired -> {
+                    performSignUpChallengeCall(
+                        oAuth2Strategy = oAuth2Strategy,
+                        signupToken = signUpStartApiResult.signupToken
+                    ).toSignUpStartCommandResult()
+                }
+                is SignUpStartApiResult.InvalidPassword -> {
+                    SignUpCommandResult.InvalidPassword(
+                        error = signUpStartApiResult.error,
+                        errorDescription = signUpStartApiResult.errorDescription
+                    )
+                }
+                is SignUpStartApiResult.InvalidAttributes -> {
+                    SignUpCommandResult.InvalidAttributes(
+                        error = signUpStartApiResult.error,
+                        errorDescription = signUpStartApiResult.errorDescription,
+                        invalidAttributes = signUpStartApiResult.invalidAttributes
+                    )
+                }
+                is SignUpStartApiResult.UsernameAlreadyExists -> {
+                    SignUpCommandResult.UsernameAlreadyExists(
+                        error = signUpStartApiResult.error,
+                        errorDescription = signUpStartApiResult.errorDescription
+                    )
+                }
+                is SignUpStartApiResult.InvalidEmail -> {
+                    SignUpCommandResult.InvalidEmail(
+                        error = signUpStartApiResult.error,
+                        errorDescription = signUpStartApiResult.errorDescription
+                    )
+                }
+                is SignUpStartApiResult.AuthNotSupported -> {
+                    SignUpCommandResult.AuthNotSupported(
+                        error = signUpStartApiResult.error,
+                        errorDescription = signUpStartApiResult.errorDescription
+                    )
+                }
+                is SignUpStartApiResult.Redirect -> {
+                    INativeAuthCommandResult.Redirect()
+                }
+                is SignUpStartApiResult.UnsupportedChallengeType, is SignUpStartApiResult.UnknownError -> {
+                    signUpStartApiResult as ApiErrorResult
+                    Logger.warn(
+                        TAG,
+                        "Unexpected result: $signUpStartApiResult"
+                    )
+                    INativeAuthCommandResult.UnknownError(
```

Comment:
I think we added a todo in the other PR for the correlationID. let's add that in this PR as well so we don't forget.